### PR TITLE
User roles assignment audit log

### PR DESCRIPTION
**Pull request by xnetsrak**
## References

* #0000

### CODE REVIEW

File: source/Energinet.DataHub.MarketParticipant.ApplyDBMigrationsApp/Scripts/LocalDB/202212131417 Add UserRoleAssignmentAuditLog Table.sql [hunk 1]
```diff
@@ -0,0 +1,23 @@
+SET ANSI_NULLS ON
+GO
```
Comment (FirestarJes) [inline source/Energinet.DataHub.MarketParticipant.ApplyDBMigrationsApp/Scripts/LocalDB/202212131417 Add UserRoleAssignmentAuditLog Table.sql:2]: This script should be added to all environments

File: source/Energinet.DataHub.MarketParticipant.Domain/Model/Users/UserRoleAssignmentChangeAuditLog.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace Energinet.DataHub.MarketParticipant.Domain.Model.Users;
+
+public enum UserRoleAssignmentTypeAuditLog
```
Comment (FirestarJes) [inline source/Energinet.DataHub.MarketParticipant.Domain/Model/Users/UserRoleAssignmentChangeAuditLog.cs:17]: Filename should match Enumname

File: source/Energinet.DataHub.MarketParticipant.Domain/Repositories/IUserRoleAssignmentAuditLogEntryRepository.cs [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Energinet.DataHub.MarketParticipant.Domain.Model.Users;
+
+namespace Energinet.DataHub.MarketParticipant.Domain.Repositories
+{
+    /// <summary>
+    /// Repository giving access to grid area audit log entries
```
Comment (FirestarJes) [inline source/Energinet.DataHub.MarketParticipant.Domain/Repositories/IUserRoleAssignmentAuditLogEntryRepository.cs:23]: Comment should be updated to reflect that this is Role Assigments and not grid areas

File: source/Energinet.DataHub.MarketParticipant.Infrastructure/Persistence/Repositories/UserRoleAssignmentAuditLogEntryRepository.cs [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    public sealed class UserRoleAssignmentAuditLogEntryRepository : IUserRoleAssignmentAuditLogEntryRepository
+    {
+        private readonly IMarketParticipantDbContext _context;
+
+        public UserRoleAssignmentAuditLogEntryRepository(IMarketParticipantDbContext context)
+        {
+            _context = context;
+        }
+
+        public Task InsertAuditLogEntryAsync(UserRoleAssignmentAuditLogEntry logEntry)
```
Comment (FirestarJes) [inline source/Energinet.DataHub.MarketParticipant.Infrastructure/Persistence/Repositories/UserRoleAssignmentAuditLogEntryRepository.cs:35]: What does this implementation do differently than the one it calls, it appears t only be a different name?

File: source/Energinet.DataHub.MarketParticipant.EntryPoint.WebApi/Controllers/UserRoleAssignmentAuditLogController.cs [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+        ILogger<UserRoleAssignmentAuditLogController> logger,
+        IUserContext<FrontendUser> userContext,
+        IMediator mediator)
+    {
+        _logger = logger;
+        _userContext = userContext;
+        _mediator = mediator;
+    }
+
+    [HttpGet("auditlogs/userroles/user/{userId:guid}/actor/{actorId:guid}")]
```
Comment (FirestarJes) [inline source/Energinet.DataHub.MarketParticipant.EntryPoint.WebApi/Controllers/UserRoleAssignmentAuditLogController.cs:46]: Seems excessive, could this be shortened a bit.

**Review by FirestarJes — CHANGES_REQUESTED** [review]

**Review by FirestarJes — APPROVED** [review]

**Review by defectiveAi — APPROVED** [review]

**Pull request merged**